virttest/qemu_devices/qdevices.py

```diff
@@ -1277,6 +1310,64 @@ def verify_unplug(self, out, monitor):
         return not self._is_attached_to_qemu(monitor)
 
 
+class QThrottleGroup(QObject):
+    """
+    throttle-group object.
+    """
+
+    def __init__(self, group_id, props):
+        self._raw_limits = props.copy()
+
+        params = QThrottleGroup._map_key(self.raw_limits)
+        params["id"] = group_id
+        kwargs = dict(backend="throttle-group", params=params)
+        super(QThrottleGroup, self).__init__(**kwargs)
+        self.throttle_group_bus = QThrottleGroupBus(group_id)
+        self.add_child_bus(self.throttle_group_bus)
+        self.set_aid(group_id)
+        self.hook_drive_bus = None
+
+    @staticmethod
+    def _map_key(params):
+        return {'x-%s' % k: v for k, v in params.items()}
+
+    @property
+    def raw_limits(self):
+        """Return raw throttle group properties."""
+        return self._raw_limits
+
+    @raw_limits.setter
+    def raw_limits(self, props):
+        """Update raw throttle group properties."""
+        self._raw_limits.update(props)
+
+    def hotplug_qmp(self):
+        """Return hotplug qmp command string."""
+        params = dict(self.params)
+        backend = params.pop("backend")
+        throttle_group_id = params.pop("id")
+        kwargs = {"qom-type": backend, "id": throttle_group_id,
+                  "props": {"limits": self.raw_limits}}
+        return "object-add", kwargs
+
+    def _query(self, monitor):
+        """Check if throttle is in use by QEMU."""
+        try:
+            monitor.qom_get(self.params["id"], "limits")
+            return True
+        except qemu_monitor.QMPCmdError as err:
+            logging.warning(err)
+            return False
+
+    def verify_hotplug(self, out, monitor):
+        """Verify if it is plugged into VM."""
+        return self._query(monitor)
+
+    def verify_unplug(self, out, monitor):
+        """Verify if it is unplugged from VM."""
+        return not self._query(monitor)
```

Comment:
The outcome will be incorrect when `self._query` executes the `except` block, we expect the method would return `False` but got `True`.

Reply:
Please check latest code

Reply:
Please double check.

Reply:
updated,please check